File: solutions/beecrowd/2031/2031.cpp
#include <cstdint>
#include <iostream>

int main() {
    int16_t n;
    std::string player1_sign, player2_sign;

    std::cin >> n;

    while (n--) {
        std::cin >> player1_sign;
        std::cin >> player2_sign;

        if (player1_sign == "ataque" && player2_sign == "ataque") {
            std::cout << "Aniquilacao mutua" << std::endl;
        } else if (player1_sign == "papel" && player2_sign == "papel") {
            std::cout << "Ambos venceram" << std::endl;
        } else if (player1_sign == "pedra" && player2_sign == "pedra") {
            std::cout << "Sem ganhador" << std::endl;
        } else if (
            player1_sign == "ataque"
            || (player1_sign == "pedra" && player2_sign == "papel")) {
            std::cout << "Jogador 1 venceu" << std::endl;
        } else {
            std::cout << "Jogador 2 venceu" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/2031/2031.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

n = int(input())

for _ in range(n):
    player_1_move = input().strip()
    player_2_move = input().strip()

    if player_1_move == 'ataque' and player_2_move == 'ataque':
        output('Aniquilacao mutua\n')
    elif player_1_move == 'papel' and player_2_move == 'papel':
        output('Ambos venceram\n')
    elif player_1_move == 'pedra' and player_2_move == 'pedra':
        output('Sem ganhador\n')
    elif (
        player_1_move == 'ataque'
        or player_1_move == 'pedra'
        and player_2_move == 'papel'
    ):
        output('Jogador 1 venceu\n')
    else:
        output('Jogador 2 venceu\n')
